Lib/idlelib/squeezer.py

```diff
@@ -46,6 +46,12 @@ def count_lines_with_wrapping(s, linewidth=80, tabwidth=8):
 
         # deal with tab or newline
         if s[pos] == '\n':
+            # If the current column was exactly linewidth, divmod would give
+            # (1,0), even though a new line hadn't yet been started. The same
+            # is true if length is any exact multiple of linewidth. Therefore,
+            # subtract 1 before dividing a non-empty line.
+            if current_column > linewidth:  # Don't bother adding 0.
+                linecount += (current_column - 1) // linewidth
```

Comment:
This code is in the tight loop of `count_lines_with_wrapping()`, which needs to be optimized, which is why I originally had this computation done all at once. The current code is indeed more readable, but the user experience is more important here IMO.

Reply:
We mostly agree but have (had) slightly different intuitions on execution time.  Pending a time test, I expect(ed) that that this *is* optimized.  The original code always had one test and two additions.  I believe that well more than 90% of the second additions will be of 0 and therefore useless.  I know that I seldom print lines longer than 80 chars.  Thinking more, where I could have been wrong is if `current_column > linewidth` uses up the time saved, relative to `current_column > 0`, because of an added subtraction.

The change is isolated in commit bd59f46 .

Reply:
Ah, I see. Let's keep it this way, then.